databricks_cli/stack/api.py

```diff
@@ -136,6 +155,82 @@ def deploy_config(self, stack_config, stack_status=None, **kwargs):
 
         return new_stack_status
 
+    def download_config(self, stack_config, **kwargs):
+        """
+        Downloads a stack given a dict of the stack configuration
+        :param stack_config: dict of stack configuration. Must contain 'resources' field.
+        :param kwargs:
+        :return:
+        """
+        self._validate_config(stack_config)
+        stack_name = stack_config.get(STACK_NAME)
+        click.echo('Downloading stack {}'.format(stack_name))
+
+        click.echo('#' * 80)
+        for resource_config in stack_config.get(STACK_RESOURCES):
+            # Deploy resource, get resource_status
+            self._download_resource(resource_config, **kwargs)
+            click.echo('#' * 80)
+
+    def _download_resource(self, resource_config, **kwargs):
+        """
+        Downloads a resource given a resource information extracted from the stack JSON
+        configuration template.
+
+        :param resource_config: A dict of the resource with fields of RESOURCE_ID, RESOURCE_SERVICE
+        and RESOURCE_PROPERTIES.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'properties': {...}}
+        """
+        resource_id = resource_config.get(RESOURCE_ID)
+        resource_service = resource_config.get(RESOURCE_SERVICE)
+        resource_properties = resource_config.get(RESOURCE_PROPERTIES)
```

Comment:
Something to consider for the future:
Operating on a dictionary (`resource_config`) is very difficult because it is hard to enforce a schema. As a result, there's a bunch of code in this class that validates and parses the data, which makes it difficult to identify what exactly the `_download_resource` function is doing.
 
Ideally, we would decouple how we parse the JSON elsewhere. By having this clear separation of concern, it'll make it easier to change the JSON structure and make the logic flow a lot more clear.